test/odbc/toby_tinyint_pg_datatype.cpp

```diff
@@ -0,0 +1,585 @@
+                          SQL_DESC_TYPE_NAME, // Get the type name of the column
+                          name,
+                          BUFFER_SIZE,
+                          NULL,
+                          NULL); 
+  // ASSERT_EQ(string(name), NAME_EXPECTED);  Same as problem above
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_int, Insertion_Success) {
+
+  const int BYTES_EXPECTED = 1;
+
+  unsigned char pk;
+  unsigned char data;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+  SQLLEN affected_rows;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  vector <string> valid_inserted_values = {
+    "0",
+    "255",
+    "3",
+    "NULL"
+  };
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN* >> bind_columns = {
+    {1, SQL_C_STINYINT, &pk, 0, &pk_len},
+    {2, SQL_C_STINYINT, &data, 0,  &data_len}
+  };
+
+  string insert_string{}; 
+  string comma{};
+  
+  for (int i = 0; i< valid_inserted_values.size(); ++i) {
+    insert_string += comma + "(" + std::to_string(i) + "," + valid_inserted_values[i] + ")";
+    comma = ",";
+  }
+
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS_TINYINT));
+  odbcHandler.CloseStmt();
+
+  // Insert valid values into the table and assert affected rows
+  odbcHandler.ExecQuery(InsertStatement(TABLE_NAME, insert_string));
+ 
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affected_rows);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(affected_rows, valid_inserted_values.size());
+  
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL1_NAME}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < valid_inserted_values.size(); ++i) {
+    
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+    ASSERT_EQ(pk_len, BYTES_EXPECTED);
+    ASSERT_EQ(pk, i);
+
+    if (valid_inserted_values[i] != "NULL")
+    {
+      ASSERT_EQ(data_len, BYTES_EXPECTED);
+      ASSERT_EQ(data, StringToTinyInt(valid_inserted_values[i]));
+    }
+    else 
+      ASSERT_EQ(data_len, SQL_NULL_DATA);
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_int, Insertion_Fail) {
+
+  const int BYTES_EXPECTED = 1;
+
+  unsigned char pk;
+  unsigned char data;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  vector <string> invalid_inserted_values = {
+    "-2147483649",
+    "2147483648"
```

Comment:
The range for `sys.tinyint` is between 0 & 255 (inclusive). 

Of course the values you have now (`-2147483649 & 2147483648`) are obviously out of range & the insertion will fail as we expect, but can we use numbers just outside the valid range instead (e.g. use -1 & 256 instead of `-2147483649 & 2147483648`, or we could have all 4 too)?